nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/test/java/org/apache/nifi/processors/cassandra/PutCassandraRecordUpdateTest.java

```diff
@@ -0,0 +1,277 @@
+                "keyField",
+                PutCassandraRecord.INCR_TYPE.getValue(),
+                Arrays.asList(
+                        new Tuple<>("keyField", 1),
+                        new Tuple<>("doubleField", 15.05D)
+                ),
+                "UPDATE keyspace.table SET doubleField=doubleField+15 WHERE keyField=1;"
+        );
+    }
+
+    @Test
+    public void testGenerateUpdateSetMultipleValues() {
+        testGenerateUpdate(
+                "keyspace.table",
+                "keyField",
+                PutCassandraRecord.SET_TYPE.getValue(),
+                Arrays.asList(
+                        new Tuple<>("keyField", 1),
+                        new Tuple<>("stringField", "newStringValue"),
+                        new Tuple<>("integerField", 15),
+                        new Tuple<>("longField", 67L)
+                ),
+                "UPDATE keyspace.table SET stringField='newStringValue',integerField=15,longField=67 WHERE keyField=1;"
+        );
+    }
+
+    @Test
+    public void testGenerateUpdateIncrementMultipleValues() {
+        testGenerateUpdate(
+                "keyspace.table",
+                "keyField",
+                PutCassandraRecord.INCR_TYPE.getValue(),
+                Arrays.asList(
+                        new Tuple<>("keyField", 1),
+                        new Tuple<>("integerField", 15),
+                        new Tuple<>("longField", 67L)
+                ),
+                "UPDATE keyspace.table SET integerField=integerField+15,longField=longField+67 WHERE keyField=1;"
+        );
+    }
+
+    @Test
+    public void testGenerateUpdateDecrementMultipleValues() {
+        testGenerateUpdate(
+                "keyspace.table",
+                "keyField",
+                PutCassandraRecord.DECR_TYPE.getValue(),
+                Arrays.asList(
+                        new Tuple<>("keyField", 1),
+                        new Tuple<>("integerField", 15),
+                        new Tuple<>("longField", 67L)
+                ),
+                "UPDATE keyspace.table SET integerField=integerField-15,longField=longField-67 WHERE keyField=1;"
+        );
+    }
+
+    private void testGenerateUpdate(String table, String updateKeys, String updateMethod, List<Tuple<String, Object>> records, String expected) {
+        Map<String, Object> recordContentMap = records.stream()
+                .collect(Collectors.toMap(Tuple::getKey, Tuple::getValue));
+
+        List<String> fieldNames = records.stream().map(Tuple::getKey).collect(Collectors.toList());
+
+        when(schema.getFieldNames()).thenReturn(fieldNames);
+        Statement actual = testSubject.generateUpdate(table, schema, updateKeys, updateMethod, recordContentMap);
+
+        assertEquals(expected, actual.toString());
+    }
+
+    private <E extends Exception> void testGenerateUpdate(String table, String updateKeys, String updateMethod, List<Tuple<String, Object>> records, E expected) {
```

Comment:
Minor: The `table` parameter is not used (it is hardcoded instead). I guess it's true that in this case it is irrelevant as we expect an exception (one that is unrelated to the `table` name).

Still we could either remove the parameter or just use it regardless. (Actually it's possible to provide an invalid table name though not sure if we consider it a valid scenario. But "keyspace." would result in an exception for example.)

Reply:
I think that moves more into the territory of determining what is valid CQL.

Actually when I was adding the code I removed the table parameter, and then re-added it, mostly for aesthetic reasons. In the future we may want to add more tests with different table names for whatever reason, and this way the usage of the testGenerateUpdate look the same regardless of expecting an exception or a CQL string as the result.